File: go/dlt645/model/data/def/variable_def.go
package def

import (
	"main/dlt645/model"
	dlt645type "main/dlt645/model/type"
)

func initVariablesDef() {
	for _, v := range dlt645type.VariableTypes {
		DIMap[uint32(v.Di)] = model.DataItem{
			Di:         uint32(v.Di),
			Name:       v.Name,
			DataFormat: v.DataFormat,
			Value:      nil,
			Unit:       v.Unit,
		}
	}
}

File: python/src/model/data/define/variable_def.py
from typing import List

from src.model.data.define.energy_def import DIMap
from src.model.types.data_type import DataItem


def init_variable_def(VariableTypes: List[DataItem]):
    for date_type in VariableTypes:
        DIMap[date_type.di] = DataItem(
            di=date_type.di,
            name=date_type.name,
            data_format=date_type.data_format,
            unit=date_type.unit
        )
